### DISCUSSION TITLE
I am trying to create plugin for counting steps for my application, but promise is not calling back method when it resolve in plugin

### DISCUSSION THREAD

**Discussion by preetamjakhmola** [ionic-team/capacitor#4843] [Q&A]
I am creating a plugin for my application in a capacitor. steps are counting and something data is coming from the android side. but my promise is not to call back in my ionic application. Do I need to implement Observable ? I tried that too but rxjs is not working in the application. In the first call, the promise is calling but when steps start counting some data logs in the app console but by promise don't call from my plugin.

My start method inside plugin
![image](https://user-images.githubusercontent.com/35592214/125986182-3d1a70b9-dc77-41e6-ab0a-6632fac8f7ed.png)

Returning data from Android resolve method
![image](https://user-images.githubusercontent.com/35592214/125987342-a84803c3-0975-4c2c-86b1-70d9c61b9688.png)

but unable to hit this method when steps are increasing.
![image](https://user-images.githubusercontent.com/35592214/125987468-735c2d1c-77c6-47c3-93e3-b1bc5bdbe587.png)

Please help with this code...

**Comment by jcesarmobile** [comment]
You are showing web plugin code, so your question doesn't make much sense if you say the data comes from Android native.

Take a look into [geolocation plugin](https://github.com/ionic-team/capacitor-plugins/tree/main/geolocation), it uses callbacks

**Reply by preetamjakhmola** [reply]
I tried to implement a callback but not much documentation is available for that. I am new to this. trying my best but something is still missing. where I can get that callback id? do I need to create a callback id for some random number and use a single callback id all time?

**Reply by jcesarmobile** [reply]
check the plugin code, you have to use `@PluginMethod(returnType = PluginMethod.RETURN_CALLBACK)` annotation and that makes the function to automatically generated the callback id and pass it to js side so you can use the id to cancel the watching whenever you want

https://github.com/ionic-team/capacitor-plugins/blob/main/geolocation/android/src/main/java/com/capacitorjs/plugins/geolocation/GeolocationPlugin.java#L86-L94
